app/lib/primer/classify/functional_colors.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Primer
+  class Classify
+    # https://primer-css-git-mkt-color-modes-docs-primer.vercel.app/css/support/v16-migration
+    class FunctionalColors
+      FUNCTIONAL_COLOR_REGEX = /(primary|secondary|tertiary|link|success|warning|danger|info|inverse|text_white)/.freeze
+
+      TEXT_COLOR_MAPPINGS = {
+        gray_dark: :text_primary,
+        gray: :text_secondary,
+        gray_light: :text_tertiary,
+        blue: :text_link,
+        green: :text_success,
+        yellow: :text_warning,
+        red: :text_danger,
+        white: :text_white,
+        # still unsure what will happen with these colors
+        black: nil,
+        orange: nil,
+        orange_light: nil,
+        purple: nil,
+        pink: nil,
+        inherit: nil
+      }.freeze
+
+      class << self
+        def text_color(val)
+          # the value is a functional color
+          return "color-#{val.to_s.dasherize}" if FUNCTIONAL_COLOR_REGEX.match?(val) || ends_with_number?(val)
```

Comment:
```suggestion
          return "color-#{val.to_s.dasherize}" if ends_with_number?(val) || FUNCTIONAL_COLOR_REGEX.match?(val) 
```

I think the number check is faster/cheaper than the regex check, thoughts on swapping the conditional around?

Reply:
I think the regex is where the additional allocations are coming from, so it's probably fine to bump up the count in the benchmark as long as it isn't increasing the cached version count.

I'd be interested in seeing what a benchmark between this and the current version of classify looks like since we now have to compare each value against a regex.

Reply:
Actually it looks like we were already running that regex, so we're probably fine benchmark wise. 👍

Reply:
I think we always compared https://github.com/primer/view_components/blob/5a71b6d309a5fba5a684cfeb52e507bed1810d00/app/lib/primer/classify.rb#L207-L208
I just moved that guard here

Reply:
> so it's probably fine to bump up the count in the benchmark as long as it isn't increasing the cached version count

I actually increased by 1, but only locally?

![image](https://user-images.githubusercontent.com/11280312/110158990-b38c5780-7daf-11eb-8478-15f38f427a2f.png)

Reply:
Hmm, what version of Ruby?

Reply:
oh, I think I'm on 2.6.3 😅

Reply:
Ah, I'm on 2.7! That's probably why.

Maybe the benchmark test should raise/not run if you aren't on the "correct" version of Ruby. 🤔